docs/cf_nats_overview.md

```diff
@@ -0,0 +1,143 @@
+# NATS in Cloud Foundry
+<!-- TOC -->
+
+- [NATS in Cloud Foundry](#nats-in-cloud-foundry)
+  - [Overview](#overview)
+  - [Architecture](#architecture)
+  - [Communication Flow](#communication-flow)
+    - [Connection Establishment](#connection-establishment)
+    - [Registeration / Unregisteration of the Application Routes in Cloud Foundry](#registeration--unregisteration-of-the-application-routes-in-cloud-foundry)
+  - [NATS Server Configurations](#nats-server-configurations)
+  - [Time Intervals and Timeouts](#time-intervals-and-timeouts)
+    - [NATS Timeouts](#nats-timeouts)
+    - [Gorouter Time Intervals](#gorouter-time-intervals)
+    - [Route Registrar Intervals](#route-registrar-intervals)
+  - [Debugging NATS in Cloud Foundry](#debugging-nats-in-cloud-foundry)
+  - [Docs](#docs)
+
+<!-- /TOC -->
+
+
+## Overview
+
+Cloud Foundry uses NATS as a publish-suscribe messaging system for route configuration. The nats-release deploys NATS as a bosh release to be used to exchange route registration/unregistration between route publishers (Route Emitters, Route Registrars) and route subscribers (Gorouters).
+## Architecture
+![](images/cf-routing.png)
+
+| Component | Description   |
```

Comment:
There are more components that connect to nats. Off the top of my head I know that the service-discovery-controller and the metrics-discovery-registrar do. 

Can you either add these to the table or indicate that this table and diagram are not an exhaustive list of all clients, but are just related to external routing and nats?